x-pack/plugins/secops/server/lib/ecs_fields/extend_map.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const extendMap = (
+  path: string,
+  map: Readonly<Record<string, string>>
+): Readonly<Record<string, string>> => {
+  const init: Record<string, string> = {};
```

Comment:
just curious: is the use of `const init: Record<string, string> = {}` here to avoid the use of `as` when declaring the initial value of reduce (like in the example below)?

```
export const extendMap = (
  path: string,
  map: Readonly<Record<string, string>>
): Readonly<Record<string, string>> =>
  Object.entries(map).reduce(
    (accum, [key, value]) => {
      accum[`${path}.${key}`] = `${path}.${value}`;
      return accum;
    },
    {} as Record<string, string>
  );
```

Reply:
Yes. From here under the Type Assertions considered harmful:
https://basarat.gitbooks.io/typescript/docs/types/type-assertion.html

Reply:
Also here is a discussion and open ticket of TypeScript's typing of array reduce in regards to why I choose the above:
https://github.com/Microsoft/TypeScript/issues/25454